cf-custom-resources/lib/wkld-cert-validator.js

```diff
@@ -0,0 +1,881 @@
+    let responseBody = JSON.stringify({
+      Status: responseStatus,
+      Reason: reasonWithLogInfo,
+      PhysicalResourceId: physicalResourceId || context.logStreamName,
+      StackId: event.StackId,
+      RequestId: event.RequestId,
+      LogicalResourceId: event.LogicalResourceId,
+      Data: responseData,
+    });
+
+    const parsedUrl = new URL(event.ResponseURL);
+    const options = {
+      hostname: parsedUrl.hostname,
+      port: 443,
+      path: parsedUrl.pathname + parsedUrl.search,
+      method: "PUT",
+      headers: {
+        "Content-Type": "",
+        "Content-Length": responseBody.length,
+      },
+    };
+
+    https
+      .request(options)
+      .on("error", reject)
+      .on("response", (res) => {
+        res.resume();
+        if (res.statusCode >= 400) {
+          reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+        } else {
+          resolve();
+        }
+      })
+      .end(responseBody, "utf8");
+  });
+}
+
+exports.handler = async function (event, context) {
+  // Destruct resource properties into local variables.
+  const props = event.ResourceProperties;
+  let { LoadBalancerDNS: loadBalancerDNS } = props;
+  const aliases = new Set(props.Aliases);
+
+  // Initialize global variables.
+  envHostedZoneID = props.EnvHostedZoneId;
+  envName = props.EnvName;
+  appName = props.AppName;
+  serviceName = props.ServiceName;
+  domainName = props.DomainName;
+  rootDNSRole = props.RootDNSRole;
+  cloudFrontCert = props.IsCloudFrontCertificate;
+  certificateDomain = cloudFrontCert
+    ? `${serviceName}.${envName}.${appName}.${domainName}`
+    : `${serviceName}-nlb.${envName}.${appName}.${domainName}`;
+  domainTypes = {
+    EnvDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${envName}.${appName}.${domainName}`),
+      domain: `${envName}.${appName}.${domainName}`,
+    },
+    AppDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${appName}.${domainName}`),
+      domain: `${appName}.${domainName}`,
+    },
+    RootDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${domainName}`),
+      domain: `${domainName}`,
+    },
+  };
+
+  let aliasesSorted = [...aliases].sort().join(",");
+  let physicalResourceID = event.PhysicalResourceId; // The certificate ARN. By default, keep old physical resource ID unchanged.
+  let handler = async function () {
+    switch (event.RequestType) {
+      case "Update":
+        let oldAliases = new Set(event.OldResourceProperties.Aliases);
+        let oldAliasesSorted = [...oldAliases].sort().join(",");
+        if (oldAliasesSorted === aliasesSorted) {
+          break;
+        }
+      // Fallthrough to "Create". When the aliases are different, the same actions are taken for both "Update" and "Create".
+      case "Create":
+        await validateAliases(aliases, loadBalancerDNS);
+        const certificateARN = await requestCertificate({
+          aliases: aliases,
+          idempotencyToken: CRYPTO.createHash("md5")
```

Comment:
what linter are you using 😢  why does it sometimes create a ton of new lines for function paraemters / promise chains and sometimes doesn't :homer-disappearing

Reply:
I use https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint 🤔

Reply:
let me try to figure it out how to make it nicer

Reply:
Actually i a prettifier extension overrides this. I increased the line length limit and hopefully i'll look nicer!